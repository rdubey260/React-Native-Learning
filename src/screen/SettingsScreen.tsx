import { useEffect, useState } from 'react';
import { View, StyleSheet, Dimensions, ActivityIndicator, TouchableOpacity, Text } from 'react-native';
import { ProgressChart } from 'react-native-chart-kit';
const screenWidth = Dimensions.get('window').width;
const data = {
  labels: ['Swim', 'Bike', 'Run'], // optional
  data: [0.4, 0.6, 0.8],
};

const chartConfig = {
  backgroundGradientFrom: '#1E2923',
  backgroundGradientFromOpacity: 0,
  backgroundGradientTo: '#08130D',
  backgroundGradientToOpacity: 0.5,
  color: (opacity = 1) => `rgba(26, 255, 146, ${opacity})`,
  strokeWidth: 2, // optional, default 3
  barPercentage: 0.5,
  useShadowColorFromDataset: false, // optional
};

interface Iprops {
  navigation: any;
}

const SettingsScreen = (props: Iprops) => {
  const [showIndicator, setShowIndicator] = useState(true);
  const { navigation } = props;

  useEffect(() => {
    runTimer()
  }, []);

  const runTimer = () => {
    setTimeout(() => {
      console.log(">>>>>")
      setShowIndicator(false);
    }, 2000);
  };

  const navigateToAxios = () => {
    navigation.navigate('AxiosScreen');
  };

  return (
    <View style={styles.container}>
      {showIndicator && <ActivityIndicator testID="activity-indicator" accessibilityHint='loading' />}
      <ProgressChart
        data={data}
        width={screenWidth}
        height={220}
        strokeWidth={16}
        radius={32}
        chartConfig={chartConfig}
        hideLegend={false}
      />
      <TouchableOpacity testID='navigate-axios' style={{ backgroundColor: 'blue' }} onPress={navigateToAxios}>
        <Text>Axios Call</Text>
      </TouchableOpacity>
    </View>
  );
};


const styles = StyleSheet.create({
  container: { flex: 1, alignItems: 'center', justifyContent: 'center' },
});

export default SettingsScreen;
