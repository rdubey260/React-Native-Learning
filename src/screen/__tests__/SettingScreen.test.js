import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SettingsScreen from '../SettingsScreen';

jest.useFakeTimers();

const navigation = {
  state: { params: {} },
  navigate: jest.fn(),
};

jest.mock('react-native-chart-kit', () => ({
  ProgressChart: () => null,
}));

test('render the settings UI', () => {
  const setting = render(<SettingsScreen />);
  const tree = setting.toJSON();
  expect(tree).toMatchSnapshot();
});

test('mock setTimeout test', () => {
  const logSpy = jest.spyOn(console, 'log');
  jest.runAllTimers();
  expect(logSpy).toBeCalledWith('>>>>>');
});


test('Navigation test', () => {
  const { getByTestId } = render(<SettingsScreen navigation={navigation} />);
  fireEvent(getByTestId('navigate-axios'), 'press');
});