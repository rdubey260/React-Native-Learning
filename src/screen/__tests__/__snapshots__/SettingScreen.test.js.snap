// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the settings UI 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityHint="loading"
    testID="activity-indicator"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "blue",
        "opacity": 1,
      }
    }
    testID="navigate-axios"
  >
    <Text>
      Axios Call
    </Text>
  </View>
</View>
`;
